worlds/sc2/client_gui.py

```diff
@@ -216,136 +224,191 @@ def build_mission_table(self, dt) -> None:
+
+                    # layout_panel.add_widget(Label(text=""))
+                    column_panel.add_widget(category_panel)
+                layout_panel.add_widget(column_panel)
+                mission_layout.add_widget(layout_panel)
             campaign_layout.add_widget(mission_layout)
             self.campaign_panel.add_widget(campaign_layout)
         self.campaign_panel.height = multi_campaign_layout_height
 
+    def mission_text(
+        self, ctx: SC2Context, mission_id: int, mission_obj: SC2Mission,
+        layout_id: int, is_layout_exit: bool, layout_name: str, campaign_id: int, is_campaign_exit: bool, campaign_name: str,
+        available_missions: List[int], available_layouts: Dict[int, List[int]], available_campaigns: List[int],
+        unfinished_missions: List[int]
+    ) -> Tuple[str, str]:
+        COLOR_MISSION_IMPORTANT = "6495ED" # blue
+        COLOR_MISSION_UNIMPORTANT = "A0BEF4" # lighter blue
+        COLOR_MISSION_CLEARED = "FFFFFF" # white
+        COLOR_MISSION_LOCKED = "A9A9A9" # gray
+        COLOR_PARENT_LOCKED = "848484" # darker gray
+        COLOR_MISSION_FINAL = "FFBC95" # orange
+        COLOR_MISSION_FINAL_LOCKED = "D0C0BE" # gray + orange
+        COLOR_FINAL_PARENT_LOCKED = "D0C0BE" # gray + orange
+        COLOR_FINAL_MISSION_REMINDER = "FF5151" # light red
+        COLOR_VICTORY_LOCATION = "FFC156" # gold
+
+        text = mission_obj.mission_name
+        tooltip: str = ""
+        remaining_locations, plando_locations, remaining_count = self.sort_unfinished_locations(mission_id)
+        parent_locked = campaign_id not in available_campaigns or layout_id not in available_layouts[campaign_id]
+
+        # Map has uncollected locations
+        if mission_id in unfinished_missions:
+            if self.any_valuable_locations(remaining_locations):
+                text = f"[color={COLOR_MISSION_IMPORTANT}]{text}[/color]"
+            else:
+                text = f"[color={COLOR_MISSION_UNIMPORTANT}]{text}[/color]"
+        elif mission_id in available_missions:
+            text = f"[color={COLOR_MISSION_CLEARED}]{text}[/color]"
+        # Map requirements not met
+        else:
+            mission_rule, layout_rule, campaign_rule = ctx.mission_id_to_entry_rules[mission_id]
+            if parent_locked:
+                text = f"[color={COLOR_PARENT_LOCKED}]{text}[/color]"
+                if campaign_id not in available_campaigns:
+                    tooltip += "To unlock this campaign, "
+                    rule_tooltip = campaign_rule.tooltip(0, lookup_id_to_mission)
+                else:
+                    tooltip += "To unlock this region, "
+                    rule_tooltip = layout_rule.tooltip(0, lookup_id_to_mission)
+            else:
+                text = f"[color={COLOR_MISSION_LOCKED}]{text}[/color]"
+                tooltip += "To unlock this mission, "
+                rule_tooltip = mission_rule.tooltip(0, lookup_id_to_mission)
```

Comment:
I'd show all the conditions that must've been met for the mission itself

Reply:
I was concerned about tooltip space with showing potentially three separate lists of requirements that will only get worse as we add more locations to missions, so I opted for this format for now.

I'm willing to budge on this if you think it'll be fine, however there's been a little bit of conversation about this in the dev chat, and I believe there's gameplay value in creating some mystery here, and some people have expressed interest in trying this out. IMO we can merge this as-is, and change it later if we get complaints about it.

Reply:
Posting here so the discord comment doesn't get lost: I think mystery is fine if the user is aware mystery is a possibility. So if there are additional requirements that you don't want to list, at least say "and additional requirements", or quality "do X to unlock this campaign/layout/mission". The screenshot Zik posted said "to unlock this region" and I think using 'region' is a really bad idea, as it's new terminology that doesn't relate to the underlying structure, and also name-collides with existing AP terminology that is different.

Reply:
I've changed this to look like this:
![python_NmzeYCCNbG](https://github.com/user-attachments/assets/db29fa4e-95c5-481e-9c1b-e3eafb57fc9b)
![python_VihMmCChzw](https://github.com/user-attachments/assets/68e0058d-c69a-4a04-a503-40737bf5ad71)